include/s3select.h

```diff
@@ -626,14 +626,16 @@ struct s3select : public bsc::grammar<s3select>
 
       cmp_operand = special_predicates | (factor) >> *(arith_cmp[BOOST_BIND_ACTION(push_compare_operator)] >> (factor)[BOOST_BIND_ACTION(push_arithmetic_predicate)]);
 
-      special_predicates = (is_null) | (is_not_null) | (between_predicate) | (in_predicate) | (like_predicate);
+      special_predicates = (is_null) | (is_not_null) | (between_predicate) | (not_between) | (in_predicate) | (like_predicate);
 
       is_null = ((factor) >> S3SELECT_KW("is") >> S3SELECT_KW("null"))[BOOST_BIND_ACTION(push_is_null_predicate)];
 
       is_not_null = ((factor) >> S3SELECT_KW("is") >> S3SELECT_KW("not") >> S3SELECT_KW("null"))[BOOST_BIND_ACTION(push_is_null_predicate)];
 
       between_predicate = (arithmetic_expression >> S3SELECT_KW("between") >> arithmetic_expression >> S3SELECT_KW("and") >> arithmetic_expression)[BOOST_BIND_ACTION(push_between_filter)];
 
+      not_between = (arithmetic_expression >> S3SELECT_KW("not") >> S3SELECT_KW("between") >> arithmetic_expression >> S3SELECT_KW("and") >> arithmetic_expression)[BOOST_BIND_ACTION(push_between_filter)];
```

Comment:
why not use a specific builder? (push_not_between_builde)

Reply:
There was a bug in the current implementation. Using a specific builder for not_between solves the issue. But, I am wondering Why not use the existing between builder similar to is null and is not null

Reply:
> There was a bug in the current implementation. Using a specific builder for not_between solves the issue. But, I am wondering Why not use the existing between builder similar to is null and is not null

good question.
the scan-statement for is-not-null is different from the "not between" 
the "not between" is a complex expression compared to "is-not-null", the scan-statement for it should be different. 

on a side note its better to refactor the is-not-null builder, it's better not to use parser-within-parser